pkg/backend/httpstate/backend.go

```diff
@@ -912,26 +912,31 @@ func (b *cloudBackend) apply(
 	}
 
 	if opts.ShowLink && !op.Opts.Display.JSONDisplay {
-		// Print a URL at the end of the update pointing to the Pulumi Service.
-		var link string
-		base := b.cloudConsoleStackPath(update.StackIdentifier)
-		if !opts.DryRun {
-			link = b.CloudConsoleURL(base, "updates", strconv.Itoa(version))
-		} else {
-			link = b.CloudConsoleURL(base, "previews", update.UpdateID)
-		}
-		if link != "" {
-			defer func() {
-				fmt.Printf(op.Opts.Display.Color.Colorize(
-					colors.SpecHeadline+"Permalink: "+
-						colors.Underline+colors.BrightBlue+"%s"+colors.Reset+"\n"), link)
-			}()
-		}
+		// Print a URL at the beginning and end of the update pointing to the Pulumi Service.
```

Comment:
Should "and end" be removed?

```suggestion
		// Print a URL at the beginning of the update pointing to the Pulumi Service.
```

Reply:
yes, thank you.